flooddrake/timestepper.py

```diff
@@ -32,10 +35,15 @@ def __init__(self, V, VCG, bed, source, Courant=0.025, func=lambda x: 1):
         # define solver
         self.v = TestFunction(V)
 
-        n = np.power((CellVolume(self.mesh).ufl_domain().topology.num_cells()/2.0),
-                     1.0/CellVolume(self.mesh).ufl_domain().geometric_dimension())
+        # Compute max lengths of each cell in DG0 function
+        Dx = MaxDx(self.mesh)
+
+        # Compute global maximum
+        dx = Dx.comm.allreduce(Dx.dat.data_ro.min(), MPI.MIN)
```

Comment:
I would pull all of this into the `MaxDx` function.  Also, did you mean `min` here, or did you want `max`?